src/Core/src/Core/Extensions/IVisualTreeElementExtensions.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Maui.Graphics;
+
+#if WINDOWS
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Media;
+using WinPoint = Windows.Foundation.Point;
+#endif
+
+namespace Microsoft.Maui
+{
+	public static class IVisualTreeElementExtensions
+	{
+		/// <summary>
+		/// Gets the Window containing the Visual Tree Element, if the element is contained within one.
+		/// </summary>
+		/// <param name="element"><see cref="IVisualTreeElement"/>.</param>
+		/// <returns><see cref="IWindow"/> if element is contained within a Window, else returns null.</returns>
+		public static IWindow? GetVisualElementWindow(this IVisualTreeElement element)
+		{
+			if (element is IWindow window)
+				return window;
+
+			var parent = element.GetVisualParent();
+			if (parent != null)
+				return parent.GetVisualElementWindow();
+
+			return null;
+		}
+
+		/// <summary>
+		/// Gets the entire list of children for a given Visual Tree Element.
+		/// </summary>
+		/// <param name="visualElement"><see cref="IVisualTreeElement"/> to scan.</param>
+		/// <returns>List of Children Elements.</returns>
+		public static IList<IVisualTreeElement> GetEntireVisualTreeElementChildren(this IVisualTreeElement visualElement) => visualElement.GetEntireVisualTreeElementChildrenInternal();
+
+		private static IList<IVisualTreeElement> GetEntireVisualTreeElementChildrenInternal(this IVisualTreeElement visualElement, IList<IVisualTreeElement>? elements = null)
+		{
+			if (elements == null)
+				elements = new List<IVisualTreeElement>();
+
+			elements.Add(visualElement);
+
+			foreach (var children in visualElement.GetVisualChildren())
+				children.GetEntireVisualTreeElementChildrenInternal(elements);
+
+			return elements;
+		}
+
+		/// <summary>
+		/// Gets list of a Visual Tree Elements children based off of a given x, y point.
+		/// </summary>
+		/// <param name="visualElement"><see cref="IVisualTreeElement"/> to scan.</param>
+		/// <param name="x1">The X point.</param>
+		/// <param name="y1">The Y point.</param>
+		/// <param name="x2">The X point.</param>
+		/// <param name="y2">The Y point.</param>
+		/// <param name="useNativeViewBounds">If else, use native view bounds for given elements. Else, use the Elements Frame.</param>
+		/// <returns>List of Children Elements.</returns>
+		public static IList<IVisualTreeElement> GetVisualTreeElements(this IVisualTreeElement visualElement, double x1, double y1, double x2, double y2, bool useNativeViewBounds = true) => GetVisualTreeElements(visualElement, new Rectangle(x1, y1, x2 - x1, y2 - y1), useNativeViewBounds);
```

Comment:
One way would be to maybe just keep rectangle as the only overload? Let that do all the checks?

Reply:
IMO, it's "Garbage in, Garbage Out".

I don't think it's worth going overboard with checks for an API like this. If you're trying to get the Visual Element Children, and you don't get the values you expect, you should be willing to check those parameters to see what's up and why it's not what you expect. If it were more forward-facing (Like something you would use for building an MAUI app) then I would be more concerned. But considering the audience for an API like this, you should be able to have some expectations of what you need to get the data to return.

If we normalize values, then it creates the expectation that you can't trust what gets passed in is actually used for calculation. 

IMO I like giving the option for a series of points rather than enforcing rectangles since then you could have hit detection based on, say, a lasso. 

But on that point, I do want to make sure that it at least returns an empty list and never blows up, I can write unit tests around that to make sure if you do give it garbage, it will return garbage ;)